conans/test/unittests/client/build/cpp_std_flags_test.py

```diff
@@ -1,195 +1,261 @@
+        self.assertEqual(make_cppstd_flag("apple-clang", "6.1", "17"), "-std=c++1z")
+
+        self.assertEqual(make_cppstd_flag("apple-clang", "7.0", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("apple-clang", "7.0", "14"), '-std=c++14')
+        self.assertEqual(make_cppstd_flag("apple-clang", "7.0", "17"), "-std=c++1z")
+
+        self.assertEqual(make_cppstd_flag("apple-clang", "8.0", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("apple-clang", "8.0", "14"), '-std=c++14')
+        self.assertEqual(make_cppstd_flag("apple-clang", "8.0", "17"), "-std=c++1z")
+
+        self.assertEqual(make_cppstd_flag("apple-clang", "9.0", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("apple-clang", "9.0", "14"), '-std=c++14')
+        self.assertEqual(make_cppstd_flag("apple-clang", "9.0", "17"), "-std=c++1z")
+
+        self.assertEqual(make_cppstd_flag("apple-clang", "9.1", "11"), '-std=c++11')
+        self.assertEqual(make_cppstd_flag("apple-clang", "9.1", "14"), '-std=c++14')
+        self.assertEqual(make_cppstd_flag("apple-clang", "9.1", "17"), "-std=c++17")
+
+        self.assertEqual(make_cppstd_flag("apple-clang", "10.0", "17"), "-std=c++17")
+        self.assertEqual(make_cppstd_flag("apple-clang", "11.0", "17"), "-std=c++17")
 
-        self.assertEqual(cppstd_flag("apple-clang", "5.1", "98"), "-std=c++98")
-        self.assertEqual(cppstd_flag("apple-clang", "5.1", "gnu98"), "-std=gnu++98")
-        self.assertEqual(cppstd_flag("apple-clang", "5.1", "11"), "-std=c++11")
-        self.assertEqual(cppstd_flag("apple-clang", "5.1", "14"), "-std=c++1y")
-
-        self.assertEqual(cppstd_flag("apple-clang", "6.1", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("apple-clang", "6.1", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("apple-clang", "6.1", "17"), "-std=c++1z")
-
-        self.assertEqual(cppstd_flag("apple-clang", "7", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("apple-clang", "7", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("apple-clang", "7", "17"), "-std=c++1z")
-
-        self.assertEqual(cppstd_flag("apple-clang", "8", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("apple-clang", "8", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("apple-clang", "8", "17"), "-std=c++1z")
-
-        self.assertEqual(cppstd_flag("apple-clang", "9", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("apple-clang", "9", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("apple-clang", "9", "17"), "-std=c++1z")
+    def test_apple_clang_cppstd_defaults(self):
+        with self.assertRaises(ConanException):
+            make_cppstd_default("apple-clang", "2.0")
+        with self.assertRaises(ConanException):
+            make_cppstd_default("apple-clang", "3.0")
+        with self.assertRaises(ConanException):
+            make_cppstd_default("apple-clang", "4.0")
```

Comment:
```suggestion
```

I would remove these ones, they are failing because they are not in the `settings.yml`, not because the `cppstd_default` function can raise.

Reply:
removed